pkg/odo/cli/component/push.go

```diff
@@ -32,6 +34,9 @@ const PushRecommendedCommandName = "push"
 // PushOptions encapsulates options that push command uses
 type PushOptions struct {
 	*CommonPushOptions
+
+	// devfile path
+	devfilePath string
```

Comment:
Why not make this a part of the `CommonPushOptions` struct in `common_push.go`?

Reply:
Because the changes in this PR do not affect the common push code path in `common_push.go`. 

In my understanding, common push has code which is common to the various `odo push` code paths whereas in the case of devfile there is common code at all.

Reply:
I think `odo push` is commong across different use cases, i.e., local sourcetype or git sourcetype or binary sourcetype. It behaves according to the underlying sourcetype. Considering DevFile is going to replace odo config (`.odo/config.yaml`) at some point, IMO, it would make more sense to make `devfilePath string` a part of `CommonPushOptions`.

Having said that, I agree that it's probably OK to keep it as-is right now. I hope we don't forget/overlook when we make it a default behaviour. :wink: